tests/helpers/test_entity_platform.py

```diff
@@ -577,6 +577,30 @@ async def test_registry_respect_entity_disabled(hass):
     assert hass.states.async_entity_ids() == []
 
 
+async def test_unique_id_conflict_has_priority_over_disabled_entity(hass, caplog):
+    """Test that an entity that is not unique has priority over a disabled entity."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    entity1 = MockEntity(
+        name="test1", unique_id="not_very_unique", enabled_by_default=False
+    )
+    entity2 = MockEntity(
+        name="test2", unique_id="not_very_unique", enabled_by_default=False
+    )
+    await component.async_add_entities([entity1])
+    await component.async_add_entities([entity2])
+
+    assert len(hass.states.async_entity_ids()) == 1
+    assert "Platform test_domain does not generate unique IDs." in caplog.text
+    assert entity1.registry_entry is not None
+    assert entity2.registry_entry is None
+    registry = er.async_get(hass)
+    # test the entity name was not updated
+    assert (
+        registry.async_get_entity_id(DOMAIN, DOMAIN, "not_very_unique")
+        == "test_domain.test1"
+    )
```

Comment:
The comment says we will check the entity name has not been updated, but it instead it checks the entity_id?

Reply:
I'll update the comment

Reply:
I think the comment was right, but the code wasn't